### PR TITLE
Add config YML param to plugin for ability to set config per job

### PR DESCRIPTION
**Pull request by mikesalvia**
PR will add the ability to set (optionally) a config path for Allure.

This is helpful in turning on and off plugins in Allure when using the Jenkins plugin.

### CODE REVIEW

**Comment by mikesalvia** [comment]
@baev Not sure who should validate this PR but I know you had mentioned something like this in and it was mentioned in #184.

**Comment by mikesalvia** [comment]
screen shot 2018-03-23 at 11 01 49 am
Got this in Jenkins... When PR was built.

**Comment by mikesalvia** [comment]
test this please

**Comment by mikesalvia** [comment]
Odd it looks like it is now running properly. Looks like there are some warnings that could be addressed.

**Comment by mikesalvia** [comment]
Anyone available to review or approve this PR?

**Comment by mikesalvia** [comment]
Still no review... :cry: (goes and hides in corner)

**Comment by baev** [comment]
Summon @eroshenkoam

**Comment by mikesalvia** [comment]
Any traction on this?

**Comment by mikesalvia** [comment]
@baev Is there another person we can poke on this?

**Comment by mikesalvia** [comment]
Bump!

**Comment by mikesalvia** [comment]
@baev Bump!

**Comment by mikesalvia** [comment]
@eroshenkoam This is the ignored PR that allows for config.yml to be added in the Jenkins plugin... I guess it can be closed.

**Comment by baev** [comment]
@mikesalvia I really appreciate your work on this, and I beg your pardon for the delay. 

In our team @eroshenkoam is responsible for the plugin, and I can't merge the change without his approval. I discussed this pull request with him several times and he doesn't like the provided changes because we are going to change plugin mechanism at some point (mystical upcoming Allure 3).

**Comment by mikesalvia** [comment]
@baev not your fault at all, I totally get it you can't merge against the maintainer's wishes, however, there was no comment at all then to see #221 it kind of got a bit frustrating to see the same work approved with no PR description or comments on why this PR was no good.

Also, it is in my opinion odd that there are no updates being done for Allure 2 because of what will happen in Allure 3. Even after Allure 3 is out people may want to (or need to) stick with Allure 2 and still have this option. Isn't that why there is version control?

**Comment by eroshenkoam** [comment]
@mikesalvia, let's make few changes here and I will merge it into master

File: src/main/java/ru/yandex/qatools/allure/jenkins/dsl/AllureReportPublisherContext.java [hunk 1]
```diff
@@ -47,4 +47,8 @@ public void property(String key, String value) {
     public void includeProperties(boolean includeProperties) {
         getPublisher().setIncludeProperties(includeProperties);
     }
+
+    public void configYml(String configYml) {
```
Comment (eroshenkoam) [inline src/main/java/ru/yandex/qatools/allure/jenkins/dsl/AllureReportPublisherContext.java]: configPath

File: src/main/java/ru/yandex/qatools/allure/jenkins/config/AllureReportConfig.java [hunk 2]
```diff
@@ -91,29 +93,35 @@ public void setIncludeProperties(Boolean includeProperties) {
...
         final AllureReportConfig config = new AllureReportConfig(results);
         config.setJdk(jdk);
         config.setCommandline(commandline);
         config.setIncludeProperties(true);
+        config.setConfigPath(configPath);
         return config;
     }
 
     private static List<ResultsConfig> convertPaths(String paths) {
+
```
Comment (eroshenkoam) [inline src/main/java/ru/yandex/qatools/allure/jenkins/config/AllureReportConfig.java]: remove this line, plz

File: src/main/java/ru/yandex/qatools/allure/jenkins/ReportBuilder.java [hunk 3]
```diff
@@ -37,26 +39,30 @@ public ReportBuilder(@Nonnull Launcher launcher, @Nonnull TaskListener listener,
...
+                                             @Nonnull FilePath reportPath, FilePath configYml)
             throws IOException, InterruptedException {
-        return version.startsWith("2") ? getAllure2Arguments(resultsPaths, reportPath)
+        return version.startsWith("2") ? getAllure2Arguments(resultsPaths, reportPath, configYml)
                 : getAllure1Arguments(resultsPaths, reportPath);
     }
 
     private ArgumentListBuilder getAllure2Arguments(@Nonnull List<FilePath> resultsPaths,
-                                                    @Nonnull FilePath reportPath) //NOSONAR
+                                                    @Nonnull FilePath reportPath, FilePath configYml) //NOSONAR
```
Comment (eroshenkoam) [inline src/main/java/ru/yandex/qatools/allure/jenkins/ReportBuilder.java]: configPath should be on new line

**Comment by eroshenkoam** [comment]
just few little changes and we are ready to merge it

File: src/main/java/ru/yandex/qatools/allure/jenkins/AllureReportPublisher.java [hunk 4]
```diff
@@ -66,6 +66,8 @@
 
     private AllureReportConfig config;
 
+    private String configYml;
```
[the remark this answers is not in the document]
Comment (mikesalvia) [reply src/main/java/ru/yandex/qatools/allure/jenkins/AllureReportPublisher.java]: 👍 done!

File: src/main/java/ru/yandex/qatools/allure/jenkins/ReportBuilder.java (hunk 3 above)
Comment (mikesalvia) [reply src/main/java/ru/yandex/qatools/allure/jenkins/ReportBuilder.java]: 👍 done!

File: src/main/java/ru/yandex/qatools/allure/jenkins/config/AllureReportConfig.java (hunk 2 above)
Comment (mikesalvia) [reply src/main/java/ru/yandex/qatools/allure/jenkins/config/AllureReportConfig.java]: 👍 done!

File: src/main/java/ru/yandex/qatools/allure/jenkins/dsl/AllureReportPublisherContext.java (hunk 1 above)
Comment (mikesalvia) [reply src/main/java/ru/yandex/qatools/allure/jenkins/dsl/AllureReportPublisherContext.java]: 👍 done!

**Comment by mikesalvia** [comment]
@eroshenkoam I have made the suggested updates. I found a bunch of other locations using `configYml`.

I also want to apologize for my reaction, it came from confusion and frustration, though it is no excuse. I understand that maintaining repositories is not an easy task, and often times things fall through and there are a lot of things that need attention, thank you for the review and working through this with me. I am happy to be able to contribute to this project!

**Comment by baev** [comment]
@mikesalvia could you please rebase from master

**Comment by mikesalvia** [comment]
@baev Re-based and force pushed! Building now.

**Comment by eroshenkoam** [comment]
ty) 👍

**Comment by eroshenkoam** [comment]
I will release it today

**Comment by timofeee** [comment]
Is this not an optional field? Updated our plugin yesterday and since we didn't specify a file we're getting "java.io.IOException: Is a directory" cause of it.

**Comment by eroshenkoam** [comment]
@timofeee sry, I fix it tomorrow

**Comment by NilsHasenbanck4758** [comment]
This just broke our jenkins. We will downgrade for now.

**Comment by Deniszen** [comment]
interim fix, adding an empty yml file to the project

**Pull request merged**